File: LC09.py
class Solution:
   def isPalindrome(self, x: int) -> bool:
        if x == 0:
            return True
        if x < 0 or x%10 == 0:
            return False
        reversed_num = 0
        while x > reversed_num:
            reversed_num = reversed_num*10 + x%10
            x /= 10
        print(x,reversed_num)
        return reversed_num == int(x) or reversed_num/10 == int(x)

solution = Solution()
solution.isPalindrome(121)
        

File: LC09.java
public class question9 {
    class Solution {
        public boolean isPalindrome(int x) {
            long num = 0;
            int temp = x;
            while (temp != 0) {
                num = num * 10 + temp % 10;
                temp /= 10;
            }
            return (int) num == x ? true : false;
        }
    }
}
